src/Illuminate/Queue/Middleware/PreventOverlappingJobs.php

```diff
@@ -0,0 +1,136 @@
+<?php
+
+namespace Illuminate\Queue\Middleware;
+
+use Illuminate\Contracts\Cache\Repository as Cache;
+
+class PreventOverlappingJobs
+{
+    /**
+     * The amount of time (in seconds) to expire the lock.
+     *
+     * @var int
+     */
+    public $expiresAt;
+
+    /**
+     * The key of the job.
+     *
+     * @var string
+     */
+    public $key;
+
+    /**
+     * The prefix of the lock key.
+     *
+     * @var string
+     */
+    public $prefix = 'overlap:';
+
+    /**
+     * The delay (in seconds) to release the job back to the queue.
+     *
+     * @var int|null
+     */
+    public $releaseAfter;
+
+    /**
+     * Create a new overlapping jobs middleware instance.
+     *
+     * @param  string  $key
+     * @param  int|null  $releaseAfter
```

Comment:
can this be null?

Reply:
Yes, it's null if we don't want to release the job afterwards. See `dontRelease` and `handle`.